server/properties.lua

```diff
@@ -1 +1,541 @@
+    end
+    TriggerClientEvent('qbx-property:client:enterGarage', source, propertyId, property.garage_slots, property.options or false)
+    Player(source).state:set('inProperty', {propertyid = propertyId}, true)
+end)
+
+--- Check for expired rents
+local function PropertiesRentCheck()
+    local rentedproperties = MySQL.query.await('SELECT * FROM properties WHERE NOT rent_expiration = false', {})
+    if not rentedproperties then return end
+
+    for _, v in pairs(rentedproperties) do
+        if v.rent_expiration / 1000 > os.time() or os.date("%x", v.rent_expiration / 1000) == os.date("%x", os.time()) then goto continue end
+        MySQL.Async.execute('UPDATE properties SET rent_expiration = NULL, garage_slots = NULL WHERE id = ?', { v.id })
+        MySQL.Async.execute('DELETE FROM property_owners WHERE property_id = ?', { v.id })
+
+        local renters = MySQL.query.await('SELECT citizenid FROM property_owners WHERE property_id = ? AND role = 1', { v.id })
+        if not renters then goto continue end
+        for _, owner in pairs(renters) do
+            TriggerEvent('qb-phone:server:sendNewMailToOffline', owner.citizenid, {
+                sender = 'Real Estate Agency',
+                subject = 'Rent Expired',
+                message = 'The rent of '.. v.name ..' expired.',
+            })
+        end
+        ::continue::
+    end
+end
+
+local function addPropertyToList(propertyData, propertyId)
+    local property = formatPropertyData(propertyData, {})
+    properties[propertyId] = property
+    updatePropertiesGroups()
+end
+
+RegisterNetEvent('qbx-property:server:CreateProperty', function(propertyData)
+    if not propertyData then return end
+    local source = source
+
+    local Player = QBCore.Functions.GetPlayer(source)
+    if not Player then return end
+
+    local PlayerData = Player.PlayerData
+    if PlayerData.job.type ~= 'realestate' then return end
+
+    local propertyId = createProperty(propertyData)
+    if not propertyId then
+        QBCore.Functions.Notify(source, Lang:t('error.failed_createproperty'), 'error')
+        return
+    end
+
+    propertyData.name = propertyId .. ' ' .. propertyData.name
+    addPropertyToList(propertyData, propertyId)
+    TriggerClientEvent('qbx-property:client:refreshProperties', -1)
+end)
+
+local function HasMoney(Player, amount)
+    if not Player then return false end
+    if Player.Functions.GetMoney('cash') >= amount then
+        return 'cash'
+    end
+    if Player.Functions.GetMoney('bank') >= amount then
+        return 'bank'
+    end
+    return false
+end
+
+--- Sets the role of a player for a property
+--- @param playerId integer
+--- @param propertyId integer
+--- @param role string
+--- @return boolean
+local function setRole(playerId, propertyId, role)
+    if not playerId or not role or not propertyId then return false end
+    local Player = QBCore.Functions.GetPlayer(playerId)
```

Comment:
capital Player?

Reply:
Isn't that how it's done throughout the framework?

Reply:
local variables should be camelCased https://qbox-docs.vercel.app/contributors#naming